crates/sui-core/src/scoring_decision.rs

```diff
@@ -49,135 +49,99 @@ pub fn update_low_scoring_authorities(
+
     if !reputation_scores.final_of_schedule {
         return;
     }
 
-    // Convert the narwhal authority ids to the corresponding AuthorityName in SUI
-    // Also capture the stake so can calculate later is strong quorum is reached for the non-low scoring authorities.
-    let scores_per_authority: HashMap<AuthorityName, (u64, Stake)> = reputation_scores
-        .scores_per_authority
-        .into_iter()
+    // We order the authorities by score ascending order in the exact same way as the reputation
+    // scores do - so we keep complete alignment between implementations
+    let scores_per_authority_order_asc: Vec<(AuthorityName, u64, Stake)> = reputation_scores
+        .authorities_by_score_desc()
+        .iter()
+        .rev() // we revert so we get them in asc order
         .map(|(authority_id, score)| {
-            let authority = committee.authority(&authority_id).unwrap();
+            let authority = committee.authority(authority_id).unwrap();
             let name: AuthorityName = authority.protocol_key().into();
 
-            // report the scores
-            if let Some(hostname) = authority_names_to_hostnames.get(&name) {
-                info!("authority {} has score {}", hostname, score);
-
-                metrics
-                    .consensus_handler_scores
-                    .with_label_values(&[&format!("{:?}", hostname)])
-                    .set(score as i64);
-            }
-
-            (name, (score, authority.stake()))
+            (name, *score, authority.stake())
         })
         .collect();
 
     let mut final_low_scoring_map = HashMap::new();
+    let mut total_stake = 0;
+    for (authority_name, score, stake) in scores_per_authority_order_asc {
+        total_stake += stake;
+
+        let included = if total_stake
+            <= (protocol_config.consensus_bad_nodes_stake_threshold() * committee.total_stake())
+                / 100 as Stake
+        {
+            final_low_scoring_map.insert(authority_name, score);
+            true
+        } else {
+            false
+        };
 
-    let mut score_list = vec![];
-    let mut nonzero_scores = vec![];
-    for (score, _stake) in scores_per_authority.values() {
-        score_list.push(*score as f64);
-        if score != &0_u64 {
-            nonzero_scores.push(*score as f64);
-        }
-    }
-
-    let median_value = median(&nonzero_scores);
-    let mut deviations = vec![];
-    let mut abs_deviations = vec![];
-    for (i, _) in score_list.clone().iter().enumerate() {
-        deviations.push(score_list[i] - median_value);
-        if score_list[i] != 0.0 {
-            abs_deviations.push((score_list[i] - median_value).abs());
-        }
-    }
+        if let Some(hostname) = authority_names_to_hostnames.get(&authority_name) {
+            info!(
```

Comment:
Is this log necessary?

Reply:
It has been useful for debug, I'll  so, good point!